lib/parachain/types.go

```diff
@@ -0,0 +1,395 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package parachain
+
+import (
+	"fmt"
+
+	"github.com/ChainSafe/gossamer/lib/common"
+	"github.com/ChainSafe/gossamer/lib/crypto/sr25519"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+)
+
+// TODO: resolve these
+// https://github.com/ChainSafe/gossamer/pull/3241#discussion_r1187883189
+// https://github.com/ChainSafe/gossamer/pull/3297#discussion_r1214740051
+
+// ValidatorIndex Index of the validator. Used as a lightweight replacement of the `ValidatorId` when appropriate
+type ValidatorIndex uint32
+
+// ValidatorID The public key of a validator.
+type ValidatorID [sr25519.PublicKeyLength]byte
+
+// BlockNumber The block number type.
+type BlockNumber uint32
+
+// GroupRotationInfo A helper data-type for tracking validator-group rotations.
+type GroupRotationInfo struct {
+	// SessionStartBlock is the block number at which the session started
+	SessionStartBlock BlockNumber `scale:"1"`
+	// GroupRotationFrequency indicates how often groups rotate. 0 means never.
+	GroupRotationFrequency BlockNumber `scale:"2"`
+	// Now indicates the current block number.
+	Now BlockNumber `scale:"3"`
+}
```

Comment:
```suggestion
type GroupRotationInfo struct {
	// SessionStartBlock is the block number at which the session started
	SessionStartBlock BlockNumber `scale:"1"`
	// GroupRotationFrequency indicates how often groups rotate. 0 means never.
	GroupRotationFrequency BlockNumber `scale:"2"`
	// Now indicates the current block number.
	Now BlockNumber `scale:"3"`
}
```
You don't need to define the `scale` struct tags if the fields are already in the right order.  I put the scale tags support in just in case we decided to re-order the attributes based for whatever reason.

Reply:
@timwu20  Were you meant to remove the scale tags in the suggested change?

Reply:
I'm just letting you know it's not required, so I didn't make the suggestion.

Reply:
haha, my bad. I thought I saw something else!